<a href="https://colab.research.google.com/github/AshleyBrooks213/DS-Unit-1-Sprint-1-Data-Wrangling-and-Storytelling/blob/master/AshleyBrooks_UNIT1Sprint2MOD4__LS_DS_124_Bayesian_Inference_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




### 1) In your own words, summarize the difference between Bayesian and Frequentist statistics.


Frequentists believe that the population number is out there somewhere, but unknown. But, there is a number that exists. 

Bayesian statistics believes that it's impossible to have a set number because it is always changing. But, we can look at what we know about our group and go from there. 

###For questions 2) - 8) use the following information:

Let A be the event of surviving the sinking of the Titanic and B be the event of being a 1st class passenger.

In [1]:
import pandas as pd
import numpy as np


data_url = 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Titanic.csv'

df = pd.read_csv(data_url, skipinitialspace=True, header=0)

print(df.shape)
df.head()

(887, 8)


Survived  Pclass  ... Parents/Children_Aboard     Fare
0         0       3  ...                       0   7.2500
1         1       1  ...                       0  71.2833
2         1       3  ...                       0   7.9250
3         1       1  ...                       0  53.1000
4         0       3  ...                       0   8.0500

[5 rows x 8 columns]

2) Calculate P(A) = P(Surving)

In [7]:
### YOUR CODE HERE

pd.crosstab(df['Sex'],df['Survived'],  margins=True)

P_A= 342/887
P_A

0.3855693348365276

3) Calculate P(B) = P(Being a 1st class passenger)/

In [10]:
### YOUR CODE HERE

pd.crosstab(df['Sex'],df['Pclass'], margins=True)

P_B = 216/887
#P_B

Pclass    1    2    3  All
Sex                       
female   94   76  144  314
male    122  108  343  573
All     216  184  487  887

4) Calculate P(A $\cap$ B) = P(Being a 1st class passenger and surviving)

In [13]:
### YOUR CODE HERE

pd.crosstab(df['Pclass'],df['Survived'], margins=True)



Survived    0    1  All
Pclass                 
1          80  136  216
2          97   87  184
3         368  119  487
All       545  342  887

In [25]:
P_A_and_B = 136/887
P_A_and_B

0.15332581736189402

5) Calculate P(A|B) = P(Surviving given that the person was a 1st class passenger)

In [17]:
### YOUR CODE HERE

P_A_given_B = P_A_and_B/P_B
print(P_A_given_B)

0.6296296296296297


6) Calculate (B|A) = P(Being a first class passenger given the person survived)

In [26]:
### YOUR CODE HERE
P_B_given_A = P_A_and_B*P_A
P_B_given_A

0.05911773341349239

7) Show that 

$$P(A|B) = \frac{P(A \cap B)}{P(B)}$$

with the numbers you calculate above.

In [27]:
### YOUR CODE HERE

P_A_given_B = P_A_and_B/P_B 

P_A_given_B

0.6296296296296297

8) Show that 

$$P(A|B) = \frac{P(B|A)P(A)}{P(B)}$$

with the numbers you calculated above.

In [28]:
### YOUR CODE HERE
P_A_given_B = (P_B_given_A*P_A)/P_B_given_A
P_A_given_B

0.3855693348365276

###Use the following information to answer questions 9) - 11) 

Assume that the passengers on the Titanic are representative of all individuals traveling from Western Europe to the United States in the early 20th century.

Let's use the Titanic data sample to estimate the mean fare paid by all individuals traveling from Western Euorpe to the United States during that time.

9) Use a 95% frequentist CI to estimate the mean fare paid by individuals traveling from Western Europe to the US in the early 20th century.

Interpret your confidence interval.

In [30]:
### YOUR CODE HERE

from scipy.stats import t

mean_fare = df['Fare'].mean()
sd_fare = df['Fare'].std()
n_fare = df['Fare'].count()
se_fare = sd_fare/(n_fare**(1/2))

t.interval(alpha = 0.95, df = n_fare-1, loc = mean_fare, scale = se_fare)

(29.02482490325031, 35.586015457516254)

Your answer here.
We are 95% confident that the  mean fare paid by individual traveling from Western Europe to the US in the early 20th century was between 29.02 and 36.

10) Use a 95% Bayesian CI to determine the distribution of the fare paid by individuals traveling from Western Europe to the US in the early 20th century.  Interpret your results

In [31]:
### YOUR CODE HERE

from scipy import stats
stats.bayes_mvs(df['Fare'], alpha=0.95)

(Mean(statistic=32.30542018038331, minmax=(29.024824903250337, 35.58601545751628)),
 Variance(statistic=2483.858450355689, minmax=(2262.747559034253, 2726.2430731483255)),
 Std_dev(statistic=49.824230556814946, minmax=(47.568346187714496, 52.213437668365835)))

We are 95% confident that the true mean fare paid by all individuals traveling from Western Europe to the US in the early 20th century can be described by a statistical distribution that has a mean of 32.30 in fare and a standard deviation of 49.82 in fare.

#Portfolio Project Assignment

Watch the Porfolio Project websites mini-lecture: https://www.youtube.com/watch?v=gxbNCGERUjc&feature=youtu.be&ab_channel=LambdaSchool


Begin to set up your own Portfolio Project website.